h2o-algos/src/main/java/hex/glm/ComputationState.java

```diff
@@ -522,7 +522,7 @@ protected boolean checkKKTs() {
       return checkKKTsMultinomial();
     double [] beta = _beta;
     double [] u = _u;
-    if(_activeData._activeCols != null) {
+    if(_activeData._activeCols != null && (_beta.length < (_dinfo.fullN()+1))) {
```

Comment:
this condition is duplicated twice, from the code is unclear when does this happen, perhaps putting this check into a method and giving it a name would clear things up

Reply:
@wendycwong can you please check if this condition is correct also for the case when the intercept is disabled (should it always be fullN + 1)?

Reply:
Honza:  Will do .

Michalk:  Yes, will check as well.

Reply:
@michalk:  dinfo.fullN() will return the number of predictors.  So, that will not change if you choose to have intercept or not.  When _parms._intercept = false, this will not change the number of coefficients.  It will still be number of predictors + intercept.  However, the intercept value will be zero.  Therefore, the condition is correct regardless of _parms._intercept = true or false.

Reply:
However, when we denormalize the coefficients, the intercept may not be zero and this is correct.

Reply:
@honzasterba:  I removed the first condition.